common/src/main/java/com/wynntils/mc/EventFactory.java

```diff
@@ -497,27 +506,34 @@ public static SetSpawnEvent onSetSpawn(BlockPos spawnPos) {
         return post(new SetSpawnEvent(spawnPos));
     }
 
-    public static void onPlayerInfoPacket(ClientboundPlayerInfoPacket packet) {
-        Action action = packet.getAction();
-        List<PlayerUpdate> entries = packet.getEntries();
+    public static void onPlayerInfoUpdatePacket(ClientboundPlayerInfoUpdatePacket packet) {
+        for (ClientboundPlayerInfoUpdatePacket.Entry entry : packet.newEntries()) {
+            GameProfile profile = entry.profile();
+            post(new PlayerLogInEvent(profile.getId(), profile.getName()));
+        }
 
-        if (action == Action.UPDATE_DISPLAY_NAME) {
-            for (PlayerUpdate entry : entries) {
-                GameProfile profile = entry.getProfile();
-                if (entry.getDisplayName() == null) continue;
-                post(new PlayerDisplayNameChangeEvent(profile.getId(), entry.getDisplayName()));
-            }
-        } else if (action == Action.ADD_PLAYER) {
-            for (PlayerUpdate entry : entries) {
-                GameProfile profile = entry.getProfile();
-                post(new PlayerLogInEvent(profile.getId(), profile.getName()));
-            }
-        } else if (action == Action.REMOVE_PLAYER) {
-            for (PlayerUpdate entry : entries) {
-                GameProfile profile = entry.getProfile();
-                post(new PlayerLogOutEvent(profile.getId()));
+        for (ClientboundPlayerInfoUpdatePacket.Entry entry : packet.entries()) {
+            for (ClientboundPlayerInfoUpdatePacket.Action action : packet.actions()) {
+                if (action == ClientboundPlayerInfoUpdatePacket.Action.UPDATE_DISPLAY_NAME) {
+                    GameProfile profile = entry.profile();
+                    if (entry.displayName() == null) continue;
+                    post(new PlayerDisplayNameChangeEvent(profile.getId(), entry.displayName()));
+                }
             }
         }
+
+        //            else if (action == Action.REMOVE_PLAYER) {
```

Comment:
@kristofbolyai  This looks like a "FIXME". I assume we still want to post PlayerLogOutEvents...

Reply:
No, I just forgot to remove this. If you look at this change, logout and other actions were separated with packets. Just below this is where I handle this.